app/serializers/serialize_community_solutions.rb

```diff
@@ -4,7 +4,7 @@ class SerializeCommunitySolutions
   initialize_with :solutions
 
   def call
-    solutions.includes(:exercise, :track).map do |solution|
+    solutions.map do |solution|
```

Comment:
Can you do two things here:
1. Add a comment here saying that `includes` occur in the searching upstream as the kaminari array is manually put together, with a warning that if this is used outside of that then, `includes(:exercise, :track)` should be added. (You could also check for the type of `solutions` and if it's an `ActiveRecord::Relation` add the includes, but the comment is probably fine for now).
2. Check that this serializer isn't called anywhere else. If it is then you'll need to do the actual work in (1) not add the comment.